codemagic.yaml

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2022 Sharezone UG (haftungsbeschränkt)
+# Licensed under the EUPL-1.2-or-later.
+#
+# You may obtain a copy of the Licence at:
+# https://joinup.ec.europa.eu/software/page/eupl
+#
+# SPDX-License-Identifier: EUPL-1.2
+
+# We are using Codemagic beside GitHub Actions for our mobile CI/CD.
+#
+# The "codemagic.yaml" allows us to configure workflows without using the
+# UI-Editor. This has the advantage that changes are tracked in Git.
+#
+# You find the documentation about the codemagic.yaml here:
+# https://docs.codemagic.io/yaml/yaml-getting-started/
+
+# With "definitions" we are able to reuse code. Therefore, we can a script once
+# and use reuse this script in other workflows without duplicating it. Just
+# define a section with &NAME and use it with *NAME.
+#
+# Docs for definitions:
+# https://docs.codemagic.io/yaml/yaml-getting-started/#reusing-sections
+#
+# Sample by Codemagic:
+# https://github.com/codemagic-ci-cd/codemagic-sample-projects/blob/main/yaml/yaml_anchors_aliases_sample/codemagic.yaml
+
+workflows:
+  app-preview:
+    name: App Preview
+    max_build_duration: 75
+    environment:
+      groups:
+        - github
+        - ios-publishing
+        - android_publishing
+        - certificate_credentials
+      vars:
+        BUNDLE_ID: de.codingbrain.sharezone.app
+      flutter: v2.10.5
+    triggering:
+      events:
+        - pull_request
+      cancel_previous_builds: true
+    working_directory: app
+    scripts:
+      - name: iOS code signing
+        script: |
+          keychain initialize
+          app-store-connect fetch-signing-files "$BUNDLE_ID" --type IOS_APP_ADHOC --create
+          keychain add-certificates
+          xcode-project use-profiles
+      - name: Set up key properties (Android)
+        script: |
+          echo $FCI_KEYSTORE | base64 --decode > android/app/key.jks
+          cat >> "android/key.properties" <<EOF
+          storePassword=$FCI_KEYSTORE_PASSWORD
+          keyPassword=$FCI_KEY_PASSWORD
+          keyAlias=$FCI_KEY_ALIAS
+          storeFile=key.jks
+          EOF
+      - name: Install dependencies
+        script: flutter pub get
+      - name: Build APK
+        script: flutter build apk --flavor dev
+      - name: Build IPA
+        script: flutter build ipa --release --export-options-plist=/Users/builder/export_options.plist --flavor dev
+    artifacts:
+      - build/**/outputs/apk/**/*.apk
+      - build/ios/ipa/*.ipa
+    publishing:
+      scripts:
+        - name: Add Dart SDK to PATH
+          script: |
+            echo PATH="$PATH":"$FLUTTER_ROOT/.pub-cache/bin" >> $CM_ENV
+            echo PATH="$PATH":"$FLUTTER_ROOT/bin" >> $CM_ENV
+        - name: Post App Preview
+          script: |
+            dart pub global activate codemagic_app_preview
+            app_preview post --gh_token $GITHUB_PAT --message "_Note_: Only Sharezone team members are able to install the iOS app."
```

Comment:
> codemagic_app_preview

They grow up so fast, now they even got their own published package we're using 😭😂